addons/account/tests/test_account_move_entry.py

```diff
@@ -1203,3 +1203,45 @@ def test_balance_modification_auto_balancing(self):
         self.assertRecordValues(line, [
             {'amount_currency': 10.00, 'balance': 10.00},
         ])
+
+    def test_error_when_different_currencies_accrued(self):
+        """
+        Tests that if two Purchase Orders with different currencies are selected for Accrued Expense Entry, 
+        a UserError is raised.
+        """
+        currency_usd = self.env['res.currency'].search([('name', '=', 'USD')], limit=1)
+        currency_eur = self.env['res.currency'].search([('name', '=', 'EUR')], limit=1)
+        if not currency_usd or not currency_eur:
+            self.skipTest("The test can not be performed without the currencies.")
```

Comment:
There are several fake currencies available, defined in common class. So instead of looking for USD and EUR, you could use two of those (try `self.company_data['currency']`,  `self.company_data_2['currency']`, `self.currency_data['currency']`, ...)